internal/elements.go

```diff
@@ -134,14 +134,17 @@ func NewOKResponse(path string) *Response {
 	}
 }
 
-func (resp *Response) Path() (string, error) {
-	if err := resp.Status.Err(); err != nil {
-		return "", err
+func (resp *Response) Path() (path string, err error) {
```

Comment:
Please don't use naked returns. Also no need for named result parameters here, it's clear what these are from the function name alone.

See https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

Reply:
No problem.
I felt it look ugly and quirky with the named returns due to the cascaded if's... or I'm not smart enough ;-)

Reply:
No named return doesn't mean you need to give up the duplicated if:

```go
err := resp.Status.Err()
var path string
if len(resp.Hrefs) == 1 {
	path = resp.Hrefs[0].Path
} else if err == nil {
	err = fmt.Errorf("webdav: malformed response: expected exactly one href element, got %v", len(resp.Hrefs))
}
return path, err
```

Reply:
Quite cool!
I definitely wasn't smart enough ;-)
My brain wasn't able to find the "if len ... else err ..." variant :-/

Thanks a lot for the simplicity lesson and sorry for the effort for such a simple pull request!